backends/arm/operators/op_add.py

```diff
@@ -51,6 +52,13 @@ def define_node(
             inputA_rescale_scale = input_A_scale.number / max_scale_2x
             inputB_rescale_scale = input_B_scale.number / max_scale_2x
 
+            broadcasted_shape = broadcast_shapes(input_A.shape, input_B.shape)
+            if permute_memory_to_nhwc:
```

Comment:
if you insert a convert node in the Graph and retrace it, it should convert all the intermediate tensors to NHWC format, can you check that so you just have to assert and not convert foreach op.

Reply:
I'm not too sure how to do this `insert a convert node in the Graph and retrace it` do you have any examples?

Reply:
Just write an `ExportPass`? There are several in XNNPACK preprocess. It should work for memory format, though I haven't tried it. Also it might break with changes Songhao is doing so that is that.

Reply:
yea, looks like a proper fix. how about let's keep my temporary solution now given the deadline and Songhao's on-going fix then we can improve given the suggestion here?